src/repository/index.ts

```diff
@@ -288,11 +295,38 @@ Message: ${err.message}`);
     this.removeAllListeners();
   }
 
+  resolveSegment(segmentId: number): Segment | undefined {
+    return this.knownSegments?.find((segment) => segment.id === segmentId);
+  }
+
+  unpackSegments(feature: FeatureInterface): FeatureInterface {
+    if (!feature?.strategies || !this.knownSegments) {
+      return feature;
+    }
+    const transformedStrategies = feature?.strategies?.map((strategy) => {
+      const segmentConstraints =
+        strategy.segments
+          ?.map((segmentId) => {
+            const foundSegment = this.resolveSegment(segmentId);
+            const constraints = foundSegment?.constraints || [];
+            return constraints;
+          })
+          ?.reduce((prev, next) => prev.concat(next), []) || [];
+      const safeConstraints = strategy?.constraints || [];
+      const safeSegmentConstraints = segmentConstraints || [];
+      return {
+        ...strategy,
+        constraints: [...safeConstraints, ...safeSegmentConstraints],
```

Comment:
This still seems potentially heavy when there are lots of segments and toggles, but not sure if too heavy.

Reply:
Yeah, I thought that too so I ran some local benchmarks. Spreading two arrays like this with a million items in each takes around 30 milliseconds (on my machine). That's probably a JSON packet measured in tens or hundreds of MB. I want to say that's probably fast enough for an excessively large packet, what do you think?

Reply:
True, that seems fine. I was thinking about memory since each segment gets copied into all its features. Maybe for later.

Reply:
That, sir, is an excellent point! And I think it's less of a "this might be heavy" and more of a "at some point this is going to be too heavy". The questions become, at what point is it too heavy, is that point within a range we want to achieve and will anything else fall over before this does. I don't have answers to those things right now, so the tradeoff I've made here is to accept that some memory balooning will happen during resolution here in exchange for much less invasive changes to the SDK. Let me setup a few benchmarks here and poke over the next few days and let some others take a look as well

Reply:
I still think we need to have some limits for segments. If we assume a segment will never contain more than, say "a few thousand items", would this be a concern? Assuming number of feature toggles are less than 1000?